olive/engine/footprint.py

```diff
@@ -99,7 +99,11 @@ def record(self, foot_print_node: FootprintNode = None, **kwargs):
         self.resolve_metrics()
 
     def get_candidates(self):
-        return {k: v for k, v in self.nodes.items() if v.metrics is not None and v.parent_model_id is not None}
+        return {
+            k: v
+            for k, v in self.nodes.items()
+            if v.metrics is not None and len(v.metrics.value) > 0 and v.parent_model_id is not None
```

Comment:
>if v.metrics is not None and len(v.metrics.value) > 0  [](http://example.com/codeflow?start=12&length=54)

if v.metrics and v.metrics.value should be more concise

Reply:
done.